packages/modal/addon/components/nucleus-modal/dialog.js

```diff
@@ -180,6 +186,56 @@ class Dialog extends Component {
     }
   }
 
+  /**
+  * createFocus
+  *
+  * @method createFocus
+  * @private
+  *
+  */
+  _createFocus() {
+    let modalDialog = this.element;
+    let focusEl = modalDialog && modalDialog.querySelector("[autofocus]");
+    if(focusEl) {
+      focusEl.focus();
+    }
+    else {
+      modalDialog.focus();
+    }
+  }
+
+  /**
+  * loopFocus
+  *
+  * @method loopFocus
+  * @private
+  * @param {any} event
+  */
+  loopFocus(event) {
+    let modalDialog = this.element.querySelector(".nucleus-modal__dialog");
+    let isTab = (event.key === 'Tab' || event.keyCode === 9);
+    if (!isTab) {
+      return;
+    }
+    let focusElements = [...modalDialog.querySelectorAll('a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input[type="text"]:not([disabled]), input[type="radio"]:not([disabled]), input[type="checkbox"]:not([disabled]), select:not([disabled])')];    
+    let currentIndex = focusElements.indexOf(document.activeElement);
+    if (currentIndex === -1) {
+      focusElements[0].focus();
+    }
+    if(event.shiftKey) {
+      if (currentIndex === 0) {
+        event.preventDefault();
+        focusElements[focusElements.length-1].focus();
+      }
+    } 
+    else {
+      if (currentIndex === (focusElements.length - 1)) {
```

Comment:
Why the nesting here? `else if`?

Reply:
Part of the residue after a previous commit where the whole keyboard mechanism was being simulated. Will cleanup.